src/main/java/baseball/game/BaseBallNumberGroup.java

```diff
@@ -0,0 +1,38 @@
+package baseball.game;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public class BaseBallNumberGroup {
+    private final List<Integer> baseBallNumbers;
+
+    public BaseBallNumberGroup(final List<Integer> baseBallNumbers) {
+        validateSize(baseBallNumbers);
+        validateDuplicate(baseBallNumbers);
+        this.baseBallNumbers = baseBallNumbers;
+    }
+
+    private void validateSize(final List<Integer> baseBallNumbers) {
+        if (baseBallNumbers.size() != GameConstants.SIZE_RANDOM_NUMBER) {
+            throw new IllegalArgumentException("숫자 야구 개수는 " + GameConstants.SIZE_RANDOM_NUMBER + "개만 가능합니다.");
+        }
+    }
+
+    private void validateDuplicate(final List<Integer> baseBallNumbers) {
+        final Set<Integer> nonDuplicateNumbers = new HashSet<>(baseBallNumbers);
+        if (nonDuplicateNumbers.size() != GameConstants.SIZE_RANDOM_NUMBER) {
+            throw new IllegalArgumentException("숫자 야구는 서로 중복될 수 없습니다.");
+        }
+    }
+
+    public int get(final int index) {
+        return baseBallNumbers.get(index);
+    }
+
+    public Stream<Integer> stream() {
+        return baseBallNumbers.stream();
+    }
```

Comment:
@leemiyeonn 님 말씀대로 값 객체를 직접 접근하게 하는 건 잘못된 것 같네요...!
아래 처럼 비즈니스 로직을 더 넣는 게 정답인 것 같습니다!!!
코드는 #1613 님꺼를 참고를 했습니다!

같은 클래스 끼리면 private 메소드에 접근이 가능하다는 것 알았네요..!
어떻게 접근해야하나 고민이었는데.. 의외로 쉽게 해결할 수 있었네요.

```suggestion
    public int getBallCount(final BaseBallNumberGroup other) {
        return (int) IntStream.range(0, baseBallNumbers.size())
                .filter(i -> other.isBall(baseBallNumbers.get(i), i))
                .count();
    }

    public int getStrikeCount(final BaseBallNumberGroup other) {
        return (int) IntStream.range(0, baseBallNumbers.size())
                .filter(i -> other.isStrike(baseBallNumbers.get(i), i))
                .count();
    }

    private boolean isStrike(int number, int digit) {
        return number == baseBallNumbers.get(digit);
    }

    private boolean isBall(int number, int digit) {
        return !isStrike(number, digit) && baseBallNumbers.contains(number);
    }
```

Reply:
반대로 저는 이걸 보고 stream을 반환하는 방법을 얻어갑니다.